sql-plugin/src/main/scala/com/nvidia/spark/rapids/CostBasedOptimizer.scala

```diff
@@ -308,6 +312,10 @@ class GpuCostModel(conf: RapidsConf) extends CostModel {
   }
 
   private def exprCost[INPUT <: Expression](expr: BaseExprMeta[INPUT], rowCount: Double): Double = {
+    if (MemoryCostHelper.isWindowExpr(expr)) {
```

Comment:
We can also make CBO best effort to minimize effect on production systems:
```scala
def exprCost(expr) = Try(doExprCost(expr)).recover {
   case _ if (!RapidsConf.isTestEnabled) => 0.0 
 // add some diagnostics logging too, or have a counter for unsuccessful cost estimates
}.get
```

Reply:
This feature is experimental and disabled by default. The implementation is likely to change a lot still so I think it might be premature for this type of hardening.